apm-agent-core/src/main/java/co/elastic/apm/agent/bci/IndyBootstrap.java

```diff
@@ -428,13 +432,19 @@ private static ConstantCallSite internalBootstrap(MethodHandles.Lookup lookup, S
                     // if config classes would be loaded from the plugin CL,
                     // tracer.getConfig(Config.class) would return null when called from an advice as the classes are not the same
                     .or(nameContains("Config").and(hasSuperType(is(ConfigurationOptionProvider.class)))));
-            Class<?> adviceInPluginCL = pluginClassLoader.loadClass(adviceClassName);
-            Class<LookupExposer> lookupExposer = (Class<LookupExposer>) pluginClassLoader.loadClass(LOOKUP_EXPOSER_CLASS_NAME);
-            // can't use MethodHandle.lookup(), see also https://github.com/elastic/apm-agent-java/issues/1450
-            MethodHandles.Lookup indyLookup = (MethodHandles.Lookup) lookupExposer.getMethod("getLookup").invoke(null);
-            // When calling findStatic now, the lookup class will be one that is loaded by the plugin class loader
-            MethodHandle methodHandle = indyLookup.findStatic(adviceInPluginCL, adviceMethodName, adviceMethodType);
-            return new ConstantCallSite(methodHandle);
+            if (addRequiredModuleOpens(requiredModuleOpens, targetClassLoader, pluginClassLoader)) {
+                Class<?> adviceInPluginCL = pluginClassLoader.loadClass(adviceClassName);
+                Class<LookupExposer> lookupExposer = (Class<LookupExposer>) pluginClassLoader.loadClass(LOOKUP_EXPOSER_CLASS_NAME);
+                // can't use MethodHandle.lookup(), see also https://github.com/elastic/apm-agent-java/issues/1450
+                MethodHandles.Lookup indyLookup = (MethodHandles.Lookup) lookupExposer.getMethod("getLookup").invoke(null);
+                // When calling findStatic now, the lookup class will be one that is loaded by the plugin class loader
+                MethodHandle methodHandle = indyLookup.findStatic(adviceInPluginCL, adviceMethodName, adviceMethodType);
+                return new ConstantCallSite(methodHandle);
+            } else {
+                // must not be a static field as it would initialize logging before it's ready
+                LoggerFactory.getLogger(IndyBootstrap.class).error("Cannot bootstrap advice because required modules could not be opened!");
+                return null;
+            }
```

Comment:
This really confused me- it looks as if the lookup handling is dependent on the module opening.
Since you already `return null`, I think it will be a clearer from this way:
```
if (!addRequiredModuleOpens(requiredModuleOpens, targetClassLoader, pluginClassLoader)) {
  // must not be a static field as it would initialize logging before it's ready
  LoggerFactory.getLogger(IndyBootstrap.class).error("Cannot bootstrap advice because required modules could not be opened!");
  return null;
}
// do the lookup exposer thing and get a method handle         
```

Furthermore, it will be even more readable if `addRequiredModuleOpens` is called when modules are supported and `!requiredModuleOpens.isEmpty()`. Then the `boolean` result returned from it really represents success or failure, which is clearer why we log an error and abort